### PR TITLE
v3 - salary and benefits page

### LINKED ISSUE #488: Render Benefits page in layout.tsx

### PR DESCRIPTION
**Pull request by mathiazom**
Closes #488

image

### CODE REVIEW

File: src/app/(main)/[slug]/page.tsx [hunk 1]
```diff
@@ -88,6 +92,13 @@ async function Page({ params }: Props) {
     );
   }
 
+  if (initialSalaryAndBenefitsPage.data) {
+    // TODO: add draft preview
```
Comment (christinaroise) [inline src/app/(main)/[slug]/page.tsx:96]: remember to also add preview 💡 

```
if(draftMode) {
return <SalaryAndBenefitsPreview initialData={initialSalaryAndBenefitsPage} />
}
```

File: src/salaryAndBenefits/SalaryAndBenefits.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import styles from "./salaryAndBenefits.module.css";
+import Text from "src/components/text/Text";
+import {SalaryAndBenefits as SalaryAndBenefitsPayload} from "studio/lib/payloads/salaryAndBenefits";
```
Comment (christinaroise) [inline src/salaryAndBenefits/SalaryAndBenefits.tsx:3]: We might want to revisit naming here.

I see blog uses BlogPage, though that be better for the component name. 🤔  Seeing as we'll add similar setups like Blog + Benefits, such as Customer cases, we should look at the naming structure. Let's tackle that in a separate PR.

File: src/salaryAndBenefits/SalaryAndBenefits.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+interface SalaryAndBenefitsProps {
+  salaryAndBenefits: SalaryAndBenefitsPayload
+}
+
+export default function SalaryAndBenefits({salaryAndBenefits}: SalaryAndBenefitsProps) {
```
Comment (christinaroise) [inline src/salaryAndBenefits/SalaryAndBenefits.tsx:9]: To keep the flexibility as well as the code conventions of the repo: 

```suggestion
const SalaryAndBenefits = ({salaryAndBenefits}: SalaryAndBenefitsProps) => {
```

and `export default SalaryAndBenefits`

File: src/salaryAndBenefits/salaryAndBenefits.module.css [hunk 4]
```diff
@@ -0,0 +1,7 @@
+.wrapper {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    align-items: center;
+    height: 800px;
```
Comment (christinaroise) [inline src/salaryAndBenefits/salaryAndBenefits.module.css:6]: remove static, you can set `min-height: 100vh` if it's absolutely needed

File: studio/lib/payloads/salaryAndBenefits.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { PortableTextBlock } from 'src/components/richText/RichText';
+
+interface Slug {
```
Comment (christinaroise) [inline studio/lib/payloads/salaryAndBenefits.ts:3]: you can use Slug from pages.ts or add global interfaces in a /payloads/gobal.ts e.g

File: src/app/(main)/[slug]/page.tsx (hunk 1 above)
Comment (mathiazom) [reply src/app/(main)/[slug]/page.tsx:96]: Assumed you wanted that in a separate PR, hence the TODO

File: src/salaryAndBenefits/SalaryAndBenefits.tsx (hunk 3 above)
Comment (mathiazom) [reply src/salaryAndBenefits/SalaryAndBenefits.tsx:9]: Just following [Mikael](https://blog.variant.no/a-better-way-to-type-react-components-9a6460a1d4b7), but consistency is nice as well 🤗

File: src/app/(main)/[slug]/page.tsx (hunk 1 above)
Comment (mathiazom) [reply src/app/(main)/[slug]/page.tsx:96]: Created #492 now, can combine with this PR if you want

Comment (christinaroise) [reply src/app/(main)/[slug]/page.tsx:96]: My bad! Overlooked the todo!

File: src/salaryAndBenefits/SalaryAndBenefits.tsx (hunk 2 above)
Comment (christinaroise) [reply src/salaryAndBenefits/SalaryAndBenefits.tsx:3]: Could you add a github task for this, and perhaps set up a quick chat about it, say tomorrow?

**Review by christinaroise — APPROVED** [review]

File: src/salaryAndBenefits/SalaryAndBenefits.tsx (hunk 2 above)
Comment (mathiazom) [reply src/salaryAndBenefits/SalaryAndBenefits.tsx:3]: Done https://github.com/varianter/variant.no/issues/493

**Pull request merged**